### PR TITLE
Merge debug and codecov, and add cuda builds

### PR DESCRIPTION
**Pull request by haampie**

### CODE REVIEW

**Comment by haampie** [comment]
bors try

**Comment by haampie** [comment]
@rasolca what tests are supposed to be run? And what about number of ranks, cpus, etc?

**Comment by rasolca** [comment]
All the tests are supposed to run. We should make sure that enabling cuda doesn't break anything.

File: ci/docker/release/build.Dockerfile [hunk 1]
```diff
@@ -100,41 +102,42 @@ RUN wget -q https://github.com/${HPX_FORK}/hpx/archive/${HPX_VERSION}.tar.gz -O
...
     cd build && \
     cmake .. \
-      -DBLASPP_BUILD_TESTS=OFF \
+      -DBUILD_TESTS=OFF \
       -DCMAKE_BUILD_TYPE=RelWithDebInfo \
       -DUSE_OPENMP=OFF \
+      -DUSE_CUDA=OFF \
       -DBLAS_LIBRARY='Intel MKL' \
       -DBLAS_LIBRARY_THREADING=sequential \
       -DCMAKE_INSTALL_PREFIX=$BLASPP_PATH && \
```
Comment (rasolca) [inline ci/docker/release/build.Dockerfile:120]: Please note that blaspp cmake has been messed up... The new config should be:
```
-Dbuild_tests=OFF
-DCMAKE_BUILD_TYPE=RelWithDebInfo
-Duse_openmp=OFF
-Dblas='Intel MKL'
-Dblas_threaded=OFF
``` 
You can see from the kubernetes output too:
```
#15 10.61 CMake Warning:
#15 10.61   Manually-specified variables were not used by the project:
#15 10.61 
#15 10.61     BLAS_LIBRARY
#15 10.61     BLAS_LIBRARY_THREADING
#15 10.61     BUILD_TESTS
#15 10.61     USE_CUDA
#15 10.61     USE_OPENMP
```

**Comment by haampie** [comment]
bors try

**Comment by haampie** [comment]
bors try-

**Comment by haampie** [comment]
bors try

**Comment by haampie** [comment]
bors try

**Comment by haampie** [comment]
bors try

**Comment by haampie** [comment]
bors try-

**Comment by haampie** [comment]
bors try

**Comment by haampie** [comment]
hopefully the last

bors try

for now :)

**Comment by haampie** [comment]
hm.

bors try-

**Comment by haampie** [comment]
bors try

**Comment by haampie** [comment]
bors ping

**Comment by haampie** [comment]
bors try

**Comment by haampie** [comment]
bors try

**Review by rasolca — APPROVED** [review]
LGTM

**Pull request merged**